dandiapi/api/services/dandiset/__init__.py

```diff
@@ -54,5 +54,7 @@ def delete_dandiset(*, user, dandiset: Dandiset) -> None:
 
     if dandiset.versions.exclude(version='draft').exists():
         raise PermissionDenied('Cannot delete dandisets with published versions.')
+    if dandiset.versions.filter(status=Version.Status.PUBLISHING).exists():
+        raise PermissionDenied('Cannot delete dandisets that are currently being published.')
```

Comment:
Technically these two checks could be done in a single query. But if we do that, from what I can tell, there wouldn't be a way to distinguish between the two cases in the response.